pypika/terms.py

```diff
@@ -101,6 +105,9 @@ def isin(self, arg):
     def notin(self, arg):
         return self.isin(arg).negate()
 
+    def as_negated(self):
```

Comment:
Please remove the negation logic from the `Criterion` class, move it up into term